AppCenterData/AppCenterData/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -189,6 +189,106 @@ - (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString
                                           fromDeviceCache:YES];
 }
 
+- (MSPaginatedDocuments *)listWithToken:(MSTokenResult *)token
+                              partition:(NSString *)partition
+                           documentType:(Class)documentType
+                            baseOptions:(MSBaseOptions *_Nullable)baseOptions {
+
+  // Final list of documents.
+  NSMutableArray<MSDocumentWrapper *> *localListItems = [NSMutableArray new];
+
+  // Get effective device time to live.
+  NSInteger deviceTimeToLive = baseOptions ? baseOptions.deviceTimeToLive : kMSDataTimeToLiveDefault;
+
+  // Execute the query.
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:token.partition];
+  NSString *selectionQuery =
+      [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\"", tableName, kMSPartitionColumnName, token.partition];
+  NSArray *listResult = [self.dbStorage executeSelectionQuery:selectionQuery];
+  NSDate *currentDate = [NSDate date];
+
+  // Parse the documents.
+  for (id documentRow in listResult) {
+
+    // If an expired document is found exclude it from the list and delete it from the local store.
+    long expirationTime = [(NSNumber *)(documentRow[self.expirationTimeColumnIndex]) longValue];
```

Comment:
`[NSNumber numberWithLong]` or `[NSNumber numberWithInt]` might suit here

Reply:
If it's represented as string then you can use `[NSString integerValue]` or `[NSString longValue]` to avoid a cast

Reply:
SQLITE_INTEGER values results in NSNumber. LGTM.